timemachine/cpp/src/exceptions.hpp

```diff
@@ -0,0 +1,7 @@
+#pragma once
+#include <stdexcept>
+
+class InvalidHardware : public std::exception {
+public:
+    const char *what() const throw() { return "Either no GPU or the GPU is acting up"; }
```

Comment:
This formatting seems weird - could `what` and `throw` be defined on separate lines?

Reply:
Switched to `const noexcept` as that is the more modern style of `const throw()` which indicates that the function doesn't raise an exception. https://github.com/proteneer/timemachine/pull/1104/commits/b0723ab266a82782e07ab114063769b08ce6e3e0

Reply:
Ah I see, was misinterpreting that as separate definitions. Makes sense.